# Addaction Analysis

Addaction have contacted us off the back of the blog on the impact on NCSO on local authorties and charities. We estimated in our blog that the impact of NCSO on Addaction is an additional £226,144.28 on top of a "usual" tariff cost of £47 257.92 giving a total of £273,402.2 in 2018. Addaction have estimated the impact on them is £403,474.59.

In [14]:
import pandas as pd
import numpy as np
from ebmdatalab import bq

In [16]:
## SQL query to extract additonal costs of NHS price concessions. 
## This is a modfied version of the SQL query from OpenPrescribing.net NHS price concession calculator site i.e. done at a "organisational level" 
sql = '''SELECT
  ncso.date AS month,
  product.bnf_code AS bnf_code,
  product.name AS product_name,
  rx.quantity AS quantity,
  rx.pct AS pct,
  ccg.org_type,
  ccg.name,
  dt.price_pence
    * rx.quantity
    * CASE WHEN
        -- For some presentations "quantity" means "number of packs" rather
        -- than e.g. tablets. In these cases we don't want to divide by the
        -- quantity value of a pack. This is implemented via a flag in our
        -- databse but this data isn't in BiqQuery so we just have a hardcoded
        -- list of BNF codes here
        product.bnf_code in ('0206010F0AACJCJ')
      THEN
        1
      ELSE
        1 / vmpp.qtyval
      END
    -- This is the "discount factor" which applies the National Average 7.2%
    -- discount to estimate Actual Cost from Net Ingredient Cost and also
    -- converts figures from pence to pounds
    * 0.00928
    AS tariff_cost,
  COALESCE(ncso.price_concession_pence - dt.price_pence, 0)
    * rx.quantity
    * CASE WHEN
        product.bnf_code in ('0206010F0AACJCJ')
      THEN
        1
      ELSE
        1 / vmpp.qtyval
      END
    * 0.00928
    AS additional_cost,
  ncso.date != rx.month AS is_projection
FROM
  dmd.ncsoconcession AS ncso
JOIN
  dmd.tariffprice AS dt
ON
  ncso.vmpp = dt.vmpp AND ncso.date = dt.date
JOIN
  dmd.product AS product
ON
  dt.product=product.dmdid
JOIN
  dmd.vmpp AS vmpp
ON
  vmpp.vppid=ncso.vmpp
JOIN
  hscic.presentation AS presentation
ON
  presentation.bnf_code = product.bnf_code
JOIN
  hscic.prescribing AS rx
ON
  rx.bnf_code = product.bnf_code
JOIN
  hscic.ccgs AS ccg #this joins to our CCG organisational data
ON
  rx.pct = ccg.code
AND
-- Where we have prescribing data for the corresponding month we use
-- that, otherwise we use the latest month of prescribing data to
-- produce an estimate
(
  rx.month = ncso.date
  OR
  (
    -- This should be set to the latest date for which we have prescribing
    -- data
    rx.month = TIMESTAMP('2019-01-01')
    AND
    ncso.date > rx.month
  )
)'''
    
df_ncso = bq.cached_read(sql, csv_path='df_ncso_nonccg.zip', compression='zip', use_cache=True)  # add `use_cache=False` to override
df_ncso.head()

TypeError: read_gbq() got an unexpected keyword argument 'compression'

In [3]:
## import "independent providers" from NHS Digital ODS codes as this is where addaction is listed.
independent_providers = pd.read_csv(r'C:\Users\bmackenna\Documents\GitHub\jupyter-notebooks\nonCCG Concessions\ephp.csv')
independent_providers.head(5)

pct                                          name national_grouping  \
0  AA4                               INTRAHEALTH LTD               Y54   
1  AA5  COMPASS WELLBEING COMMUNITY INTEREST COMPANY               Y56   
2  AA6                  ASSISTED CONCEPTION UNIT LTD               Y56   
3  AA7                         LEYLAND PHYSIOTHERAPY               Y54   
4  AA8                                     SMART CJS               Y59   

  high_level_health_geography                   address line 1  \
0                         Q74  1ST FLOOR, WILLIAM BROWN CENTRE   
1                         Q71        STEELS LANE HEALTH CENTRE   
2                         Q71                LEYTONSTONE HOUSE   
3                         Q84                      83 BOW LANE   
4                         Q87                       BUILDING B   

                    Unnamed: 5               Unnamed: 6  Unnamed: 7  \
0                    MANOR WAY                      NaN    PETERLEE   
1      384-388 COMMERCIAL ROAD                      NaN      LONDON   
2                  LEYTONSTONE                      NaN      LONDON   
3                          NaN                      NaN     LEYLAND   
4  KIRTLINGTON BUSINESS CENTRE  SLADE FARM, KIRTLINGTON  KIDLINGTON   

       Unnamed: 8  postcode  open_date  close_date  
0   COUNTY DURHAM   SR8 5TW   20130401         NaN  
1  GREATER LONDON    E1 0LR   20130401         NaN  
2  GREATER LONDON   E11 1GA   20130401         NaN  
3      LANCASHIRE  PR25 4YB   20130401         NaN  
4     OXFORDSHIRE   OX5 3JA   20130401         NaN

In [4]:
## adding in independent provider names to our prescribing dataset result from SQL
indopro = df_ncso.merge(independent_providers[['pct','name']],  how="outer", on='pct')
indopro.head()

month         bnf_code              product_name  quantity  pct  \
0 2018-08-01  0704020J0AAACAC    Oxybutynin 5mg tablets     210.0  10A   
1 2018-08-01  1304000Y0AAAAAA     Mometasone 0.1% cream     480.0  10A   
2 2018-08-01  0408010H0AAACAC  Lamotrigine 25mg tablets     678.0  10A   
3 2018-08-01  040201030AAAAAA   Risperidone 1mg tablets     716.0  10A   
4 2018-08-01  0206030N0AAAAAA   Nicorandil 10mg tablets     600.0  10A   

  org_type                    name_x  tariff_cost  additional_cost  \
0      CCG  NHS SOUTH KENT COAST CCG     4.872000         7.830000   
1      CCG  NHS SOUTH KENT COAST CCG    26.414592         8.775168   
2      CCG  NHS SOUTH KENT COAST CCG    14.381349        51.907680   
3      CCG  NHS SOUTH KENT COAST CCG    20.265664       110.962816   
4      CCG  NHS SOUTH KENT COAST CCG    17.075200        21.715200   

  is_projection name_y  
0         False    NaN  
1         False    NaN  
2         False    NaN  
3         False    NaN  
4         False    NaN

In [5]:
## clearing blanks from the datset
df2 = indopro.dropna(subset=['name_y'])
df2

month         bnf_code  \
9616108 2018-02-01  0501011P0AAASAS   
9616109 2018-03-01  0501011P0AAARAR   
9616110 2016-06-01  0403010X0AAADAD   
9616111 2018-01-01  0403030D0AAABAB   
9616112 2018-10-01  0204000R0AAAJAJ   
9616113 2018-08-01  0408010W0AAADAD   
9616114 2018-05-01  0704020N0AAABAB   
9616115 2018-11-01  0501012G0AAABAB   
9616116 2017-12-01  0408010AEAAAGAG   
9616117 2017-12-01  0408010G0AAABAB   
9616118 2017-12-01  0304010E0AAAAAA   
9616119 2018-07-01  1304000Y0AABABA   
9616120 2018-01-01  0906027G0AAABAB   
9616121 2018-02-01  0501011P0AAARAR   
9616122 2017-10-01  0403030D0AAAAAA   
9616123 2017-09-01  0403040Y0AAABAB   
9616124 2018-11-01  1001010P0AAADAD   
9616125 2017-11-01  0408010G0AAAAAA   
9616126 2017-11-01  0403030D0AAAAAA   
9616127 2017-11-01  0403040Y0AAAAAA   
9616128 2018-11-01  0205052N0AAADAD   
9616129 2017-10-01  040702040AAAAAA   
9616130 2017-05-01  0403040X0AAANAN   
9616131 2019-03-01  0402010ABAAAKAK   
9616132 2018-08-01  0407041T0AAAFAF   
9616133 2017-07-01  0407041T0AAAFAF   
9616134 2017-07-01  0407041T0AAACAC   
9616135 2017-04-01  0403010B0AAAIAI   
9616136 2018-06-01  0501080W0AAAEAE   
9616137 2019-02-01  0501110C0AAAJAJ   
...            ...              ...   
9925974        NaT              NaN   
9925975        NaT              NaN   
9925976        NaT              NaN   
9925977        NaT              NaN   
9925978        NaT              NaN   
9925979        NaT              NaN   
9925980        NaT              NaN   
9925981        NaT              NaN   
9925982        NaT              NaN   
9925983        NaT              NaN   
9925984        NaT              NaN   
9925985        NaT              NaN   
9925986        NaT              NaN   
9925987        NaT              NaN   
9925988        NaT              NaN   
9925989        NaT              NaN   
9925990        NaT              NaN   
9925991        NaT              NaN   
9925992        NaT              NaN   
9925993        NaT              NaN   
9925994        NaT              NaN   
9925995        NaT              NaN   
9925996        NaT              NaN   
9925997        NaT              NaN   
9925998        NaT              NaN   
9925999        NaT              NaN   
9926000        NaT              NaN   
9926001        NaT              NaN   
9926002        NaT              NaN   
9926003        NaT              NaN   

                                              product_name  quantity  pct  \
9616108  Phenoxymethylpenicillin 250mg/5ml oral solutio...    1000.0  NDL   
9616109  Phenoxymethylpenicillin 125mg/5ml oral solutio...    1800.0  NDL   
9616110                            Trazodone 150mg tablets      28.0  NDL   
9616111                            Citalopram 10mg tablets     168.0  NDL   
9616112                           Propranolol 40mg tablets     280.0  NDL   
9616113    Sodium valproate 500mg gastro-resistant tablets      56.0  NDL   
9616114                            Tolterodine 2mg tablets      56.0  NDL   
9616115                      Flucloxacillin 500mg capsules      56.0  NDL   
9616116                          Pregabalin 300mg capsules      56.0  NDL   
9616117                          Gabapentin 300mg capsules     268.0  NDL   
9616118                         Fexofenadine 120mg tablets     150.0  NDL   
9616119                           Mometasone 0.1% ointment      30.0  NDL   
9616120                  Vitamin B compound strong tablets     112.0  NDL   
9616121  Phenoxymethylpenicillin 125mg/5ml oral solutio...     600.0  NDL   
9616122                            Citalopram 20mg tablets     812.0  NDL   
9616123          Duloxetine 60mg gastro-resistant capsules      28.0  NDL   
9616124                             Naproxen 250mg tablets    1764.0  NDL   
9616125                          Gabapentin 100mg capsules    1020.0  NDL   
9616126                            Citalopram 20mg tablets     280.0  NDL   
9616127          Duloxetine 30mg gastro-resistant caps

In [6]:
## filtering the dataset to only include results where Addaction or similar is in the independent provider name
addaction2 = df2[df2["name_y"].str.contains("addaction", case=False)]
addaction2.head()

month         bnf_code  \
9917793 2017-03-01  0403040X0AAAAAA   
9917794 2017-02-01  0403040X0AAAAAA   
9917795 2016-11-01  0407041M0AAAAAA   
9917796 2018-10-01  0410030A0AAAEAE   
9917797 2018-10-01  0410030A0AAADAD   

                                            product_name  quantity  pct  \
9917793                         Mirtazapine 30mg tablets      91.0  NI3   
9917794                         Mirtazapine 30mg tablets      98.0  NI3   
9917795                        Naratriptan 2.5mg tablets       6.0  NI3   
9917796  Buprenorphine 8mg sublingual tablets sugar free     832.0  NI3   
9917797  Buprenorphine 2mg sublingual tablets sugar free     343.0  NI3   

        org_type name_x  tariff_cost  additional_cost is_projection     name_y  
9917793  Unknown   None     3.830320         1.025440         False  ADDACTION  
9917794  Unknown   None     4.124960         1.104320         False  ADDACTION  
9917795  Unknown   None     1.883840        20.898560         False  ADDACTION  
9917796  Unknown   None   236.040777      1462.570423         False  ADDACTION  
9917797  Unknown   None    43.198400       195.529600         False  ADDACTION

In [7]:
## Counting how many variations on the name Addaction are retured
addaction2["name_y"].unique()

array(['ADDACTION'], dtype=object)

There is only one variation of the name Addaction in our dataset therefore we can be reasonably confident that our impact is correct as per the NHS BSA primary care prescribing/reimbursement dataset.

In [8]:
### see impact of NCSO in 2018
addaction2018 = addaction2.loc[(addaction2["month"]>="2018-01-01") & (addaction2["month"]<="2018-12-01")]
addaction2018

month         bnf_code  \
9917796 2018-10-01  0410030A0AAAEAE   
9917797 2018-10-01  0410030A0AAADAD   
9917798 2018-09-01  0410030A0AAADAD   
9917799 2018-10-01  0410030A0AAADAD   
9917800 2018-05-01  0410030A0AAAEAE   
9917802 2018-04-01  0410030A0AAAEAE   
9917803 2018-04-01  0410030A0AAADAD   
9917804 2018-07-01  0410030A0AAAEAE   
9917805 2018-07-01  0410030A0AAAEAE   
9917806 2018-04-01  0410030A0AAAEAE   
9917807 2018-04-01  0410030A0AAADAD   
9917808 2018-08-01  0410030A0AAAEAE   
9917809 2018-07-01  0410030A0AAADAD   
9917811 2018-04-01  0410030A0AAAEAE   
9917812 2018-02-01  0206020A0AAABAB   
9917813 2018-09-01  0410030A0AAADAD   
9917814 2018-09-01  0410030A0AAAEAE   
9917815 2018-12-01  0410030A0AAADAD   
9917816 2018-12-01  0410030A0AAAEAE   
9917820 2018-06-01  0410030A0AAAEAE   
9917821 2018-06-01  0410030A0AAAEAE   
9917822 2018-03-01  0501011P0AAARAR   
9917823 2018-12-01  0410030A0AAADAD   
9917824 2018-11-01  0410030A0AAADAD   
9917834 2018-05-01  0403040W0AAABAB   
9917835 2018-07-01  0403040W0AAABAB   
9917836 2018-01-01  0403040W0AAABAB   
9917840 2018-07-01  040201030AAAKAK   
9917844 2018-09-01  0410030A0AAAEAE   
9917845 2018-05-01  0410030A0AAADAD   
...            ...              ...   
9918138 2018-08-01  0410030A0AAADAD   
9918139 2018-08-01  0410030A0AAAEAE   
9918140 2018-08-01  0410030A0AAAEAE   
9918143 2018-11-01  0410030A0AAADAD   
9918144 2018-06-01  0410030A0AAADAD   
9918145 2018-04-01  0410030A0AAADAD   
9918146 2018-07-01  0410030A0AAAEAE   
9918147 2018-07-01  0410030A0AAADAD   
9918148 2018-11-01  0401010T0AAAMAM   
9918149 2018-11-01  0205052N0AAABAB   
9918150 2018-10-01  0410030A0AAAEAE   
9918151 2018-10-01  0410030A0AAADAD   
9918152 2018-09-01  0410030A0AAAEAE   
9918154 2018-03-01  0501011P0AAASAS   
9918155 2018-07-01  0410030A0AAADAD   
9918156 2018-07-01  0410030A0AAADAD   
9918157 2018-08-01  0410030A0AAAEAE   
9918158 2018-09-01  0410030A0AAADAD   
9918159 2018-09-01  0410030A0AAADAD   
9918160 2018-06-01  0410030A0AAADAD   
9918161 2018-06-01  0410030A0AAAEAE   
9918162 2018-12-01  0401010T0AAAMAM   
9918163 2018-12-01  0410030A0AAAEAE   
9918164 2018-01-01  0408010AEAAAAAA   
9918166 2018-06-01  0410030A0AAADAD   
9918167 2018-05-01  0410030A0AAAEAE   
9918169 2018-04-01  0201010F0AAAEAE   
9918184 2018-01-01  0906027G0AAABAB   
9918185 2018-03-01  0906027G0AAABAB   
9918186 2018-02-01  0906027G0AAABAB   

                                              product_name  quantity  pct  \
9917796    Buprenorphine 8mg sublingual tablets sugar free     832.0  NI3   
9917797    Buprenorphine 2mg sublingual tablets sugar free     343.0  NI3   
9917798    Buprenorphine 2mg sublingual tablets sugar free    3615.0  NI3   
9917799    Buprenorphine 2mg sublingual tablets sugar free     396.0  NI3   
9917800    Buprenorphine 8mg sublingual tablets sugar free     910.0  NI3   
9917802    Buprenorphine 8mg sublingual tablets sugar free     951.0  NI3   
9917803    Buprenorphine 2mg sublingual tablets sugar free    8365.0  NI3   
9917804    Buprenorphine 8mg sublingual tablets sugar free    5446.0  NI3   
9917805    Buprenorphine 8mg sublingual tablets sugar free     481.0  NI3   
9917806    Buprenorphine 8mg sublingual tablets sugar free      68.0  NI3   
9917807    Buprenorphine 2mg sublingual tablets sugar free     163.0  NI3   
9917808    Buprenorphine 8mg sublingual tablets sugar free    1124.0  NI3   
9917809    Buprenorphine 2mg sublingual tablets sugar free     690.0  NI3   
9917811    Buprenorphine 8mg sublingual tablets sugar free     105.0  NI3   
9917812                            Amlodipine 10mg tablets      28.0  NI3   
9917813    Buprenorphine 2mg sublingual tablets sugar free    8391.0  NI3   
9917814    Buprenorphine 8mg sublingual tablets sugar free     546.0  NI3   
9917815    Buprenorphine 2mg sublingual tablets sugar free     955.0  NI3   
9917816    Buprenorphine 8mg sublingual tablets sugar free     868.0  NI3   
9917820    Buprenorphine 8mg sublingual tablets sugar 

In [9]:
### check impact for 2018
addaction2018.sum()

quantity           278206.000000
tariff_cost         47257.922650
additional_cost    226144.279588
dtype: float64

Now we will check the spend on Buprenorphine (BNF code = 0410030A0%) to see if Addaction are counting total impact instead of diferentiating between NCSO and "normal cost". The difference here is we are looking at all Buprenorphine products in all months of 2018. The previous analysis only look at products in months when they are NCSO.

In [11]:
sql = '''SELECT
  extract(year from month) AS year,
  bnf_code AS bnfcode,
  bnf_name AS bnfname,
  SUM(items) AS total_items,
  SUM(quantity) AS total_qty,
  SUM(actual_cost) AS actual_cost
FROM
  ebmdatalab.hscic.normalised_prescribing_standard AS presc
WHERE
  bnf_code LIKE "0410030A0%"  
  and
  pct = "NI3" 
GROUP BY
  year,
  bnf_code,
  bnf_name'''
    
df_addaction = bq.cached_read(sql, csv_path='df_addactionbuprenorphine.csv', use_cache=True)  # add `use_cache=False` to override
df_addaction.head()

C:\Users\bmackenna\Anaconda3\lib\site-packages\pandas_gbq\gbq.py:556: FutureWarning: verbose is deprecated and will be removed in a future version. Set logging level in order to vary verbosity
  "verbosity", FutureWarning, stacklevel=1)


year          bnfcode                                   bnfname  \
0  2012  0410030A0AAADAD  Buprenorphine_Tab Subling 2mg S/F          
1  2017  0410030A0BBABAD                   Subutex_Tab Subling 2mg   
2  2015  0410030A0AAADAD  Buprenorphine_Tab Subling 2mg S/F          
3  2010  0410030A0AAAAAA  Acamprosate Calc_Tab E/C 333mg             
4  2019  0410030A0AAADAD         Buprenorphine_Tab Subling 2mg S/F   

   total_items  total_qty  actual_cost  
0          708      24602  24035.80000  
1          843      22031  18634.76891  
2         8384     205332  59216.95000  
3            1         28      3.72000  
4          710      18859  16819.16657

In [12]:
### grouping the total cost by year
totalcost = df_addaction.groupby(['year'])['actual_cost'].sum()
totalcost

year
2010         3.72000
2011     53178.14000
2012    330436.05000
2013    315208.23000
2014    333620.98000
2015    247939.29000
2016    215990.79000
2017    171170.42258
2018    352194.52391
2019     51633.75501
Name: actual_cost, dtype: float64

In [13]:
## graphing full years - i.e. get rid of 2010 and 2019 as incomplete
totalcostfullyearsonly = totalcost.drop([2010, 2019])
ax = totalcostfullyearsonly.plot(kind='line', title = "Total Cost £ of Buprenorphine at Addaction")
ax.set_ylim(0, 400000)


(0, 400000)

Addaction spent £352,194.52 on Buprenorphine  (BNF code = 0410030A0%) as per the prescribing file which is £51,280 short of their estimate of £403,474.59. We will now clarify/ask
-  What is the time frame? 18/19 has not finished so are they estimating or is the £403k a partial year spend?
-  What is the datset? for example is this invoiced?
-  Do they have other prescribing codes where the name is not similar to Addaction?
-  Do they pay through a "recharge" mechanism? e.g. a CCG pays and then invoices addaction

Addaction have responded to say they have done an estimate for 2018/2019 and that our calculation are in line with their estimates.